bin/run_epix

```diff
@@ -1,122 +1,133 @@
+
+
+def main(args, return_df=False):
+    # TODO: remove setup from main for strax
+    path, file_name, detector, outer_cylinder = setup(args)
+
+    if args.debug:
+        print("epix configuration: ", args)
+        # TODO: also add memory information see starxer and change this to debug
+        # Getting time information:
         starttime = time.time()
         tnow = starttime
-    # Loading the data:
-    p = args.InputFile
-    p = p.split('/')
-    if p[0] == "":
-        p[0] = "/"
-    path = os.path.join(*p[:-1])
-    file_name = p[-1]
-    print(f'Reading in root file {file_name}')
-    inter = epix.loader(path, file_name, kwargs_uproot_ararys={'entry_stop': args.EntryStop})
-    if args.Timing:
-        t = time.time()
-        print(f'It took {round(t - tnow, 5)} sec to load the data.')
-        tnow = t
-
-    print((f'Finding clusters of interactions with a dr = {args.MicroSeparation} mm'
-           f' and dt = {args.MicroSeparationTime} ns'))
+
+    # Loading data:
+    inter = epix.loader(path, file_name, args.debug,
+                        outer_cylinder=outer_cylinder,
+                        kwargs_uproot_arrays={'entry_stop': args.EntryStop}
+                        )
+
+    if args.debug:
+        tnow = monitor_time(tnow, 'load data.')
+        print((f'Finding clusters of interactions with a dr = {args.MicroSeparation} mm'
+               f' and dt = {args.MicroSeparationTime} ns'))
+
+    # Cluster finding and clustering:
     inter = epix.find_cluster(inter, args.MicroSeparation/10, args.MicroSeparationTime)
-    if args.Timing:
-        t = time.time()
-        print(f'It took {round(t - tnow, 5)} sec to find clusters.')
-        tnow = t
-        
-    result = epix.cluster(inter, args.TagClusterBy=='energy')
-    if args.Timing:
-        t = time.time()
-        print(f'It took {round(t - tnow, 5)} sec to cluster events.')
-        tnow = t
+
+    if args.debug:
+        tnow = monitor_time(tnow, 'cluster finding.')
+
+    result = epix.cluster(inter, args.TagClusterBy == 'energy')
+
+    if args.debug:
+        tnow = monitor_time(tnow, 'cluster merging.')
 
     # Add eventid again:
     result['evtid'] = ak.broadcast_arrays(inter['evtid'][:, 0], result['ed'])[0]
-    
+
     # Sort detector volumes and keep interactions in selected ones:
-    sensitive_volumes = [epix.tpc, epix.below_cathode] #TODO: add options
-    print('Removing clusters not in volumes:', *[x.name for x in sensitive_volumes])
-    print(f'Number of clusters before: {np.sum(ak.num(result["x"]))}')
-    result['ids'] = epix.in_sensitive_volume(result, sensitive_volumes)
-    m = (result['ids'] == sensitive_volumes[0].volume_id) | (result['ids'] == sensitive_volumes[1].volume_id)
-    # TODO: The idea is to code this properly and depending on len(sensitive_volumes)
+    if args.debug:
+        print('Removing clusters not in volumes:', *[v.name for v in detector])
+        print(f'Number of clusters before: {np.sum(ak.num(result["ed"]))}')
```

Comment:
The `Removing clusters not in volumes` output isn't entirely true, since we then list all the volumes defined for a detector in the detectors file, and not these in which we set the `to_be_stored` option to true.

Reply:
`detectors` should only contain the volumes which have `to_be_stored=True`

Reply:
I get the list of all implemented volumes, when loading the `example_xenonnt.ini` file (which has two volumes set to `False`).

Reply:
Okay, I was setting the old config file name. This means epix doesn't complain if you give it a config filename that doesn't exist under `--Config`.

Reply:
Okay solved, also added test for empty data/no interactions left.